demo/app/tests/internal/tasks/schedulers/time-based/android/power-savings-manager.android.spec.ts

```diff
@@ -1,39 +1,57 @@
 import { PowerSavingsManager } from "nativescript-task-dispatcher/internal/tasks/schedulers/time-based/android/alarms/power-savings-manager.android";
+import {createOsForegroundActivityMock} from "~/tests/internal/tasks/schedulers/time-based/android/index";
+import {Utils} from "@nativescript/core";
+import {
+    createSavingsDeactivationIntent
+} from "nativescript-task-dispatcher/internal/tasks/schedulers/time-based/android/intents.android";
 
 describe("Power savings manager", () => {
     if (typeof android === "undefined") {
         return;
     }
 
-    const osPowerManager = createOsPowerManagerMock();
+    const powerManagerMock = createOsPowerManagerMock();
+    const foregroundActivityMock = createOsForegroundActivityMock();
 
     beforeEach(() => {
-        spyOn(
-            osPowerManager,
-            "isIgnoringBatteryOptimizations"
-        ).and.callThrough();
+        spyOn(powerManagerMock, "isIgnoringBatteryOptimizations").and.returnValue(false);
+        spyOn(foregroundActivityMock, "startActivity");
     });
 
     it("checks if power savings are enabled", () => {
-        const powerSavingsManager = new PowerSavingsManager(osPowerManager);
+        const powerSavingsManager = new PowerSavingsManager(powerManagerMock);
 
         const areDisabled = powerSavingsManager.areDisabled();
 
-        expect(areDisabled === true || areDisabled === false).toBeTruthy();
+        expect(areDisabled).toBeFalse();
```

Comment:
This test only works if tested on Android 23+ devices, and minSdk is 22

Reply:
Done, I knew that strange equality check had a purpose, but didn't know which one hehe

Reply:
I've decided to add SDK version checks instead of this weird thing, so this does not happen again 😅